### PR TITLE
CUMULUS-965 Remove cwait

### PR DESCRIPTION
**Pull request by yjpa7145**
We are using a number of different packages to handle throttling of async operations.  After discussion, decided to use the p-* libraries provided here: https://github.com/sindresorhus/promise-fun

### CODE REVIEW

File: packages/common/aws.js [hunk 1]
```diff
@@ -226,15 +226,14 @@ exports.s3CopyObject = (params) => {
...
-  return concurrency.toPromise(uploadFn, params);
-};
+ * Upload data to S3
+ *
+ * Note: This is equivalent to calling `aws.s3().upload(params).promise()`
+ *
+ * @param {Object} params - see [S3.upload()](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property)
+ * @returns {Promise} see [S3.upload()](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property)
+ */
+exports.promiseS3Upload = (params) => exports.s3().upload(params).promise();
```
Comment (markdboyd) [inline packages/common/aws.js]: So we're intentionally removing the concurrency limit here?

File: packages/common/aws.js [hunk 2]
```diff
@@ -617,11 +616,6 @@ class DynamoDbScanQueue {
 }
 exports.DynamoDbScanQueue = DynamoDbScanQueue;
 
-exports.syncUrl = async (uri, bucket, destKey) => {
```
Comment (markdboyd) [inline packages/common/aws.js]: Is this no longer needed?

File: packages/common/concurrency.js [hunk 3]
```diff
@@ -44,32 +62,11 @@ const toPromise = (fn, ...args) =>
  * @param {function} condition - A function which determines whether fn is called.
  * @param {function} fn - The function to call if condition returns true
  * @param {*} args - Arguments to pass to calls to both condition and fn
- * @returns {Promise<*>} - A promise that resolves to either null or the result of fn
+ * @returns {Promise} A promise that resolves to either null or the result of fn
 */
 const unless = (condition, fn, ...args) =>
   Promise.resolve((condition(...args) ? null : fn(...args)));
 
-const promiseUrl = (urlstr) =>
```
Comment (markdboyd) [inline packages/common/concurrency.js]: Is this no longer used/needed?

**Comment by markdboyd** [comment]
This PR is good to go pending @yjpa7145 getting confirmation that the `promiseUrl` function isn't still used by GITC

**Comment by yjpa7145** [comment]
Breaking this apart into three separate PRs.

**Pull request closed without merging**